tests/index.spec.ts

```diff
@@ -0,0 +1,224 @@
+import Big from "big.js";
+import MockDate from "mockdate";
+import { ComitMetadata } from "../src/comitNode";
+import { BitcoinConfig, Config } from "../src/config";
+import Balances from "../src/rates/balances";
+import TestnetMarketMaker from "../src/rates/testnetMarketMaker";
+import { TradeService } from "../src/rates/tradeService";
+import { getAmountsToPublishRoute } from "../src/routes/tradesToPublish";
+import EthereumWalletStub from "./doubles/ethereumWalletStub";
+
+function getBitcoinConfig(): BitcoinConfig {
+  const config = Config.fromFile("./tests/configs/testnetMarketMaker.toml");
+  if (!config.bitcoinConfig) {
+    throw new Error("Could not load bitcoin config");
+  }
+  return config.bitcoinConfig;
+}
+
+const ethereumWalletMock = new EthereumWalletStub({
+  nominalBalance: new Big(1000),
+  chainId: 17
+});
+
+function balance(balance: number) {
+  return Promise.resolve(new Big(balance));
+}
+
+const comitMetadata = {
+  id: "somePeerId",
+  listenAddresses: ["/ip4/127.0.0.1/tcp/8011"]
+} as ComitMetadata;
+
+describe("Bobtimus API tests", () => {
+  it("Should return the trade with the same timestamp if the quantities did not change", async () => {
+    const bitcoinBalance = 100;
+    const etherBalance = 1000;
+
+    const balances = await Balances.create(
+      {
+        bitcoin: () => balance(bitcoinBalance),
+        ether: () => balance(etherBalance)
+      },
+      20
+    );
+
+    const testnetMarketMaker: TradeService = new TestnetMarketMaker(
+      { rateSpread: 5, publishFraction: 200, maxFraction: 100 },
+      await balances
+    );
+
+    const apiCall = getAmountsToPublishRoute(
+      testnetMarketMaker,
+      getBitcoinConfig(),
+      ethereumWalletMock,
+      comitMetadata
+    );
+
+    const req = {
+      body: {}
+    };
+
+    const res = {
+      sendCalledWith: "",
+      async send(arg: any) {
+        this.sendCalledWith = arg;
+      }
+    };
+
+    const expected = {
+      peerId: "somePeerId",
+      addressHint: comitMetadata.listenAddresses[0],
+      ledgers: [
+        {
+          name: "bitcoin",
+          network: "regtest"
+        },
+        {
+          name: "ethereum",
+          network: "regtest"
+        }
+      ],
+      rates: [
+        {
+          timestamp: "Sun, 14 Jun 2015 22:12:05 GMT",
```

Comment:
📣 format should be a RFC 3339 timestamp https://github.com/coblox/makeshift/pull/2/commits/9060118ea40bfc594cc9b81746d9b20bf7814779